extensions/src/main/java/org/pentaho/platform/web/http/api/resources/services/FileService.java

```diff
@@ -777,6 +778,13 @@ public RepositoryFileToStreamWrapper doGetFileOrDir( String pathId ) throws File
     return wrapper;
   }
 
+  public void doesFileExists( String destPathId ) throws FileNotFoundException {
+    RepositoryFileDto repositoryFileDto = getRepoWs().getFile( idToPath( destPathId ) );
+    if ( Objects.isNull( repositoryFileDto ) ) {
```

Comment:
unit test for FileService#doesFileExists

Reply:
Have entirely removed the FileService.java code, have leveraged an existing method doesExists.